python/pyarrow/_dataset_parquet.pyx

```diff
@@ -301,7 +301,11 @@ cdef class ParquetFileFragment(FileFragment):
 
     def __reduce__(self):
         buffer = self.buffer
-        row_groups = [row_group.id for row_group in self.row_groups]
+        if not bool(self.parquet_file_fragment.row_groups()):
```

Comment:
Can you maybe add a comment to clarify that if `parquet_file_fragment.row_groups()` is empty, this means that the metadata information is not yet populated